rust/c509-certificate/src/lib.rs

```diff
@@ -13,27 +17,27 @@
 //! # TBS Certificate
 //!
 //! The To Be Sign Certificate contains the following fields:
-//!    * c509CertificateType: A certificate type, whether 0 a natively signed C509
-//!      certificate following X.509 v3 or 1 a CBOR re-encoded X.509 v3 DER certificate.
+//!    * c509CertificateType: A certificate type, whether 2 a natively signed C509
+//!      certificate following X.509 v3 or 3 a CBOR re-encoded X.509 v3 DER certificate.
 //!    * certificateSerialNumber: A unique serial number for the certificate.
-//!    * issuer: The entity that issued the certificate.
+//!    * subjectPublicKeyAlgorithm: The algorithm that the public key is used.
+//!    * issuer: The entity that issued the certificate. In case off self-signed, the
+//!      issuer is identical to the subject.
```

Comment:
subjectPublicKeyAlgorithm: Specifies the cryptographic algorithm used for the public key, including its encoding and operational parameters.
issuer: The entity that issued the certificate. In the case of  a self-signed certificate, the issuer is identical to the subject.

Reply:
I think this part 'including its encoding and operational parameters.' might be too detail which can cause confusion